### PR TITLE
filebeat/module/mysql: add authentication message parsing and related.ip & related.user fields

### LINKED ISSUE #32231: Filebeat MySQL ingest pipeline can't parse authentication messages for SIEM
**Describe the enhancement:**

Filebeat MySQL error ingest pipeline can't parse failed authentication message.
ECS fields important for SIEM like event.category, event.type, event.action, event.outcome and source.ip, user.name are missing.

Failed authentication message should have fields:

- event.category = "authentication"
- event.action = "logon-failed"
- event.type = "start"
- event.outcome = "failure"
- source.ip
- user.name

**Describe a specific use case for the enhancement or feature:**

```
2022-07-06T01:30:35.766704Z 230349 [Note] Access denied for user 'root'@'10.10.10.10' (using password: YES)
2022-07-06T01:30:35.766704Z 230349 [Note] Access denied for user 'root'@'localhost' (using password: YES)

```
**Steps to  reproduce**

```
POST /_ingest/pipeline/filebeat-7.17.4-mysql-error-pipeline/_simulate
{
  "docs": [
    {
      "_source": {
        "@timestamp": "2022-07-06T01:30:35.766Z",
        "message": "2022-07-06T01:30:35.766704Z 230349 [Note] Access denied for user 'root'@'10.10.10.10' (using password: YES)"
      }
    }
  ]
}
```

Result

```
{
  "docs" : [
    {
      "doc" : {
        "_index" : "_index",
        "_type" : "_doc",
        "_id" : "_id",
        "_source" : {
          "@timestamp" : "2022-07-06T01:30:35.766Z",
          "log" : {
            "level" : "Note"
          },
          "mysql" : {
            "thread_id" : 230349
          },
          "message" : "Access denied for user 'root'@'10.10.10.10' (using password: YES)",
          "event" : {
            "ingested" : "2022-07-06T16:30:27.331184518Z",
            "category" : [
              "database"
            ],
            "type" : [
              "info"
            ],
            "created" : "2022-07-06T01:30:35.766Z",
            "kind" : "event"
          }
        },
        "_ingest" : {
          "timestamp" : "2022-07-06T16:30:27.331184518Z"
        }
      }
    }
  ]
}
```

### LINKED ISSUE #34634: Filebeat MySQL slowlog module - user.name exists and related.user field missing
**Describe the enhancement:**

Filebeat MySQL slowlog module missing related.user field. 

According to [field description](https://www.elastic.co/guide/en/ecs/current/ecs-related.html#field-related-user):

> All the user names or other user identifiers seen on the event

The problem not only with this one module. Seems like here is no autotest to check related.user in case user.name field exists.
Created [feature request](https://github.com/elastic/beats/issues/32663) to add the autotest for this case.

**Describe a specific use case for the enhancement or feature:**

Check [mariadb-10.3.13.log-expected.json](https://github.com/elastic/beats/blob/main/filebeat/module/mysql/slowlog/test/mariadb-10.3.13.log-expected.json) - user.name field exists and related.user missing.

### PR DESCRIPTION
**Pull request by leweafan**
## What does this PR do?

Adds new patterns, authentication message parsing and related.ip & related.user fields

## Why is it important?

Parsing should be ECS compliant and authentication messages important for SIEM

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #34634
- Closes #32231

### CODE REVIEW

**Comment by leweafan** [comment]
Can any one help with review?

File: filebeat/module/mysql/error/ingest/pipeline.yml [hunk 1]
```diff
@@ -67,6 +69,49 @@ processors:
...
+    target_field: source.as.number
+    ignore_missing: true
+- rename:
+    field: source.as.organization_name
+    target_field: source.as.organization.name
+    ignore_missing: true
+- append:
+    field: related.ip
+    value: "{{source.ip}}"
+    if: ctx?.source?.ip != null
```
Comment (efd6) [inline filebeat/module/mysql/error/ingest/pipeline.yml:95]:
```suggestion
    if: ctx.source?.ip != null
```

File: filebeat/module/mysql/error/ingest/pipeline.yml [hunk 2]
```diff
@@ -67,6 +69,49 @@ processors:
...
+- append:
+    field: related.user
+    value: "{{user.name}}"
+    if: ctx?.user?.name != null
```
Comment (efd6) [inline filebeat/module/mysql/error/ingest/pipeline.yml:99]:
```suggestion
    if: ctx.user?.name != null
```

File: filebeat/module/mysql/error/ingest/pipeline.yml [hunk 3]
```diff
@@ -67,6 +69,49 @@ processors:
...
+- append:
+    field: event.category
+    value: authentication
+    if: ctx?._tmp?.auth_failed != null
```
Comment (efd6) [inline filebeat/module/mysql/error/ingest/pipeline.yml:103]:
```suggestion
    if: ctx._tmp?.auth_failed != null
```

File: filebeat/module/mysql/error/ingest/pipeline.yml [hunk 4]
```diff
@@ -67,6 +69,49 @@ processors:
...
+- append:
+    field: event.action
+    value: logon-failed
+    if: ctx?._tmp?.auth_failed != null
```
Comment (efd6) [inline filebeat/module/mysql/error/ingest/pipeline.yml:107]:
```suggestion
    if: ctx._tmp?.auth_failed != null
```

File: filebeat/module/mysql/error/ingest/pipeline.yml [hunk 5]
```diff
@@ -67,6 +69,49 @@ processors:
...
+- set:
+    field: event.outcome
+    value: failure
+    if: ctx?._tmp?.auth_failed != null
```
Comment (efd6) [inline filebeat/module/mysql/error/ingest/pipeline.yml:111]:
```suggestion
    if: ctx._tmp?.auth_failed != null
```

File: filebeat/module/mysql/error/test/mysql-ubuntu-5.5.53-extented.log-expected.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "@timestamp": "2023-03-12T19:26:00.000-02:00",
+        "event.category": [
+            "database"
```
Comment (efd6) [inline filebeat/module/mysql/error/test/mysql-ubuntu-5.5.53-extented.log-expected.json:5]: Is this event also 'network'?

File: filebeat/module/mysql/slowlog/ingest/pipeline.json [hunk 7]
```diff
@@ -94,6 +94,20 @@
         "field": "event.type",
         "value": "info"
       }
+    },
+    {
+      "append": {
+        "field": "related.user",
+        "value": "{{user.name}}",
+        "if": "ctx?.user?.name != null"
```
Comment (efd6) [inline filebeat/module/mysql/slowlog/ingest/pipeline.json:102]:
```suggestion
        "if": "ctx.user?.name != null"
```

File: filebeat/module/mysql/slowlog/ingest/pipeline.json [hunk 8]
```diff
@@ -94,6 +94,20 @@
...
+      }
+    },
+    {
+      "append": {
+        "field": "related.ip",
+        "value": "{{source.ip}}",
+        "if": "ctx?.source?.ip != null"
```
Comment (efd6) [inline filebeat/module/mysql/slowlog/ingest/pipeline.json:109]:
```suggestion
        "if": "ctx.source?.ip != null"
```

File: CHANGELOG.next.asciidoc [hunk 9]
```diff
@@ -232,6 +232,7 @@ automatic splitting at root level, if root level element is an array. {pull}3415
 - Add Basic Authentication support on constructed requests to CEL input {issue}34609[34609] {pull}34689[34689]
 - Add string manipulation extensions to CEL input {issue}34610[34610] {pull}34689[34689]
 - Add unix socket log parsing for nginx ingress_controller {pull}34732[34732]
+- Add MySQL authentication message parsing and related.ip & related.user fields {pull}34810[34810]
```
Comment (efd6) [inline CHANGELOG.next.asciidoc:235]:
```suggestion
- Add MySQL authentication message parsing and `related.ip` and `related.user` fields {pull}34810[34810]
```

**Review by efd6 — COMMENTED** [review]
Please also resolve the merge conflict.

File: filebeat/module/mysql/error/test/mysql-ubuntu-5.5.53-extented.log-expected.json (hunk 6 above)
Comment (leweafan) [reply filebeat/module/mysql/error/test/mysql-ubuntu-5.5.53-extented.log-expected.json:5]: I suppose it is and maybe connection too

Comment (efd6) [reply filebeat/module/mysql/error/test/mysql-ubuntu-5.5.53-extented.log-expected.json:5]: In `event.type`, yeah, probably.

Comment (leweafan) [reply filebeat/module/mysql/error/test/mysql-ubuntu-5.5.53-extented.log-expected.json]: fixed pipeline and added network event.category. event.type is info cause connection require source.ip and destination.ip

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**